lib/ansible/modules/network/fortios/fortios_ipv4_policy.py

```diff
@@ -0,0 +1,324 @@
+        id                        = dict(type='str', required=True),
+        src_intf                  = dict(default='any'),
+        dst_intf                  = dict(default='any'),
+        state                     = dict(choices=['present', 'absent'], default='present'),
+        src_addr                  = dict(required=True, type='list'),
+        dst_addr                  = dict(required=True, type='list'),
+        src_addr_negate           = dict(type='bool', default=False),
+        dst_addr_negate           = dict(type='bool', default=False),
+        action                    = dict(choices=['accept','deny'], required=True),
+        service                   = dict(aliases=['services'], required=True, type='list'),
+        service_negate            = dict(type='bool', default=False),
+        schedule                  = dict(type='str', default='always'),
+        nat                       = dict(type='bool', default=False),
+        fixedport                 = dict(type='bool', default=False),
+        poolname                  = dict(type='str'),
+        av_profile                = dict(type='str'),
+        webfilter_profile         = dict(type='str'),
+        ips_sensor                = dict(type='str'),
+        application_list          = dict(type='str'),
+    )
+
+    #merge global required_if & argument_spec from module_utils/fortios.py
+    argument_spec.update(fortios_argument_spec)
+    required_if = fortios_required_if
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=required_if,
+    )
+
+    result = dict(changed=False)
+
+    # fail if libs not present
+    msg = ""
+    if not HAS_PYFG:
+        module.fail_json(msg='Could not import the python library pyFG required by this module')
+
+    #define device
+    f = FortiOS( module.params['host'],
+        username=module.params['username'],
+        password=module.params['password'],
+        timeout=module.params['username'],
+        vdom=module.params['vdom'])
+
+    path = 'firewall policy'
+
+    #connect
+    try:
+        f.open()
+    except:
+        module.fail_json(msg='Error connecting device')
+
+    #get  config
+    try:
+        f.load_config(path=path)
+        result['firewall_address_config'] = f.running_config.to_text()
+    except:
+        module.fail_json(msg='Error reading running config')
+
+    #Absent State
+    if module.params['state'] == 'absent':
+        f.candidate_config[path].del_block(module.params['id'])
+        change_string = f.compare_config()
+        if change_string != "":
+            result['change_string'] = change_string
+            result['changed'] = True
+
+    #Present state
+    if module.params['state'] == 'present':
+        new_policy = FortiConfig(module.params['id'], 'edit')
+
+        #src / dest / service / interfaces
+        new_policy.set_param('srcintf', '"{0}"'.format(module.params['src_intf']))
```

Comment:
My first daft was without "_" but I made lot of errors in playbook, I think it's more readable this way